src/pybind/mgr/dashboard_v2/tools.py

```diff
@@ -114,6 +118,128 @@ def mgr(self):
+                    self._view.exception = None
+
+            self.event.set()
+
+    class RemoteViewCache(object):
+
+        def __init__(self, timeout):
+            self.getter_thread = None
+            # Consider data within 1s old to be sufficiently fresh
+            self.stale_period = 1.0
+            # Return stale data if
+            self.timeout = timeout
+            self.event = threading.Event()
+            self.value_when = None
+            self.value = None
+            self.latency = 0
+            self.exception = None
+            self.lock = threading.Lock()
+
+        def run(self, fn, args, kwargs):
+            """
+            If data less than `stale_period` old is available, return it
+            immediately.
+            If an attempt to fetch data does not complete within `timeout`, then
+            return the most recent data available, with a status to indicate that
+            it is stale.
+
+            Initialization does not count towards the timeout, so the first call
+            on one of these objects during the process lifetime may be slower
+            than subsequent calls.
+
+            :return: 2-tuple of value status code, value
+            """
+            with self.lock:
+                now = datetime.datetime.now()
+                if self.value_when and now - self.value_when < datetime.timedelta(
+                        seconds=self.stale_period):
+                    return ViewCache.VALUE_OK, self.value
+
+                if self.getter_thread is None:
+                    self.getter_thread = ViewCache.GetterThread(self, fn, args,
+                                                                kwargs)
+                    self.getter_thread.start()
+
+                ev = self.getter_thread.event
+
+            success = ev.wait(timeout=self.timeout)
+
+            with self.lock:
+                if success:
+                    # We fetched the data within the timeout
+                    if self.exception:
+                        # execution raised an exception
+                        return ViewCache.VALUE_EXCEPTION, self.exception
+                    return ViewCache.VALUE_OK, self.value
+                elif self.value_when is not None:
+                    # We have some data, but it doesn't meet freshness requirements
+                    return ViewCache.VALUE_STALE, self.value
+                # We have no data, not even stale data
+                return ViewCache.VALUE_NONE, None
+
+    def __init__(self, timeout=5):
+        self.timeout = timeout
+        self.cache_by_args = {}
+
+    def __call__(self, fn):
+        def wrapper(*args, **kwargs):
+            rvc = self.cache_by_args.get(args, None)
+            if not rvc:
+                rvc = ViewCache.RemoteViewCache(self.timeout)
+                self.cache_by_args[args] = rvc
+            return rvc.run(fn, args, kwargs)
+        return wrapper
+
+
+class CephPoolConnection(object):
+    _connections = {}
```

Comment:
Here's the thing though: we are adding elements to this dictionary each time we open an ioctx to a given pool that hasn't been cached yet.
While we can assume a production cluster won't have an infinite number of pools, it's not completely out of the question if test clusters, or PoCs, or whatever, may end up creating and removing pools for testing purposes. Even a production cluster may have temporary pools for some reason.
All the while, we're adding elements to this dictionary and I haven't seen those entries being removed. This is a cache that may grow unbounded until the mgr is restarted (or the module un/reloaded, I'm guessing).
Maybe the memory footprint will not be a problem. Regardless, not having a criteria to prune old/unused entries seems like an anti-pattern for a cache.

Reply:
I think you're right, I will need to think a bit about this problem and how to solve it

Reply:
Just to clarify, this problem also exists on legacy dashboard, right?

Reply:
@ricardoasmarques yes, I think so

Reply:
Cache-invalidation. What about just keeping ioctx just for 5 mins?

Reply:
@sebastian-philipp that should work.